libass/ass.c

```diff
@@ -114,6 +114,10 @@ int ass_alloc_style(ASS_Track *track)
 
     sid = track->n_styles++;
     memset(track->styles + sid, 0, sizeof(ASS_Style));
+
+    // Explicitly flag newer values as unset
+    track->styles[sid].MarginB = INT_MIN;
```

Comment:
This in-band magic value does make things simpler and avoids the per-style flag field from #651 required to deal with the tricky “API-created style not setting MarginB + user style overrides for MarginT” case. On the other hand, the magic “unset” constant occupies a technically valid input value (and currently this PR does not deal with style overrides at all)

Perhaps, we just shouldn’t allow users to override `MarginB`/`MarginT` individually on pre-v4++ files and decide purely by `track_type` what to do with bottom margins? In this case we’d need neither an in-band magic value nor the flag field iinm

Reply:
I don't think INT_MIN is a meaningfully useful value here. I considered shifting a parsed INT_MIN to INT_MIN+1, but it's very difficult to imagine a real use-case (or even an accidental situation where it actually matters). I suppose with a bit of tweaking, I could have `INT_MIN` only signal "no value" when `track_type` is pre-ASS2 (which would effectively mean that `INT_MIN` is allowed in files but invalid in overrides), but I'm not sure how worthwhile that'd be.